src/codegen/compile.ml

```diff
@@ -10455,20 +10579,20 @@ and compile_dec env pre_ae how v2en dec : VarEnv.t * G.t * (VarEnv.t -> scope_wr
       unmodified
     )
 
-  | VarD (name, _, e) ->
+  | VarD (name, typ, e) ->
```

Comment:
Good that you mention this. I indeed assumed otherwise - and fixed it. Although, luckily, it does not matter for barrier optimization since it only considers the type without distinguishing Mut or not (because it knows it is already a write at those locations). But it would be a hard-to-find bug and pitfall if the type would be used for other purposes in future.